app/web/src/components/IdeToolbarNew/IdeToolbarNew.js

```diff
@@ -80,6 +83,63 @@ const IdeToolbarNew = ({ cadPackage }) => {
       copyTextToClipboard(window.location.href)
     }
   }
+  const PullTitleFromFirstLine = (code) => {
+    const firstLine = code.split('\n').filter(identity)[0]
+    if (!(firstLine.startsWith('//') || firstLine.startsWith('#'))) {
+      return 'object.stl'
+    }
+    return (
+      (firstLine.replace(/^(\/\/|#)\s*(.+)/, (_, __, titleWithSpaces) =>
+        titleWithSpaces.replaceAll(/\s/g, '-')
+      ) || 'object') + '.stl'
+    )
+  }
+
+  const handleStlDownload = (({ geometry, fileName, type }) => () => {
+    const makeStlBlobFromGeo = flow(
+      (geo) => new Mesh(geo, new MeshBasicMaterial()),
+      (mesh) => new Scene().add(mesh),
+      (scene) => new STLExporter().parse(scene),
+      (stl) =>
+        new Blob([stl], {
+          type: 'text/plain',
+        })
+    )
+    const saveFile = (geometry) => {
+      const blob = makeStlBlobFromGeo(geometry)
+      fileSave(blob, {
+        fileName,
+        extensions: ['.stl'],
+      })
+    }
+    if (geometry) {
+      if (type === 'geometry') {
```

Comment:
if we're currently showing 3d in the preview than we're golden, if not than it must be openscad, so we should trigger an stl download.